Taurine/exploit/kfd/libkfd/krkw.h

```diff
@@ -208,6 +208,10 @@ void krkw_helper_run_allocate(struct kfd* kfd, struct krkw* krkw)
         for (u64 i = 0; i < kfd->puaf.number_of_puaf_pages; i++) {
             u64 puaf_page_uaddr = kfd->puaf.puaf_pages_uaddr[i];
             print_buffer(puaf_page_uaddr, 64);
+            
+            uint32_t magicToSearch = 0x1EA5CACE;
+            void *res = memmem((void *)kfd->puaf.puaf_pages_uaddr[i], 0x4000, &magicToSearch, sizeof(magicToSearch));
+            assert(res == NULL);
```

Comment:
```suggestion
        struct utsname systemInfo;
	uname(&systemInfo);
	if (strncmp(systemInfo.machine, "iPad13",6) == 0 && strcmp(systemInfo.machine, "iPad13,1") != 0 && strcmp(systemInfo.machine, "iPad13,2") != 0) {
            for (u64 i = 0; i < kfd->puaf.number_of_puaf_pages; i++) {
                u64 puaf_page_uaddr = kfd->puaf.puaf_pages_uaddr[i];
                print_buffer(puaf_page_uaddr, 64);
            
                 uint32_t magicToSearch = 0x1EA5CACE;
                 void *res = memmem((void *)kfd->puaf.puaf_pages_uaddr[i], 0x4000, &magicToSearch, sizeof(magicToSearch));
                 assert(res == NULL);
            }
        } else {
            for (u64 i = 0; i < kfd->puaf.number_of_puaf_pages; i++) {
                u64 puaf_page_uaddr = kfd->puaf.puaf_pages_uaddr[i];
                print_buffer(puaf_page_uaddr, 64);
            }
        }
```
not completely sure if this would break M1, but just to be sure for now I'm keeping the check in.